Jenkinsfile

```diff
@@ -319,13 +334,162 @@ def job = {
                     sh "docker push ${config.dockerRegistry}${dockerRepo}:master-latest"
 
                     if (config.release) {
-                        sh "docker tag ${config.dockerRegistry}${dockerRepo}:${config.docker_tag} ${config.dockerRegistry}${dockerRepo}:${config.docker_tag}-${env.BUILD_NUMBER}"
-                        sh "docker push ${config.dockerRegistry}${dockerRepo}:${config.docker_tag}-${env.BUILD_NUMBER}"
+                        // docker_tag for release builds does not include the build number
+                        // here we add it back so an image version with the build number always exists
+                        sh "docker tag ${config.dockerRegistry}${dockerRepo}:${config.docker_tag} ${config.dockerRegistry}${dockerRepo}:${config.docker_tag}-beta${env.BUILD_NUMBER}"
+                        sh "docker push ${config.dockerRegistry}${dockerRepo}:${config.docker_tag}-beta${env.BUILD_NUMBER}"
                     }
                 }
             }
         }
     }
+    }
+
+    // Build CCloud ksqlDB image
+
+    if (config.release) {
+        // Do not build CCloud image in release mode
+        return null
+    }
+
+    stage('Checkout CCloud ksqlDB Docker repo') {
+        checkout changelog: false,
+            poll: false,
+            scm: [$class: 'GitSCM',
+                branches: [[name: config.ccloud_revision]],
+                doGenerateSubmoduleConfigurations: false,
+                extensions: [[$class: 'RelativeTargetDirectory', relativeTargetDir: 'ksqldb-ccloud-docker']],
+                submoduleCfg: [],
+                userRemoteConfigs: [[credentialsId: 'ConfluentJenkins Github SSH Key',
+                    url: 'git@github.com:confluentinc/cc-docker-ksql.git']]
+            ]
+    }
+
+    stage('Build CCloud KSQL Docker image') {
+        dir('ksqldb-ccloud-docker') {
+            archiveArtifacts artifacts: 'pom.xml'
+            withCredentials([
+                usernamePassword(credentialsId: 'JenkinsArtifactoryAccessToken', passwordVariable: 'ARTIFACTORY_PASSWORD', usernameVariable: 'ARTIFACTORY_USERNAME'),
+                usernameColonPassword(credentialsId: 'Jenkins GitHub Account', variable: 'GIT_CREDENTIAL')]) {
+                    withDockerServer([uri: dockerHost()]) {
+                        withMaven(globalMavenSettingsFilePath: settingsFile, options: mavenOptions) {
```

Comment:
@vcrfxia this is your issue right here. this is still configured to point at the beta maven repo. up above where it creates the settingsFile, it configures it to the beta repo. so if you are just hacking something together, create the file again but change it to the regular repo where all the snapshots are

Reply:
Discussed offline: turns out it's not possible to use both the default Jenkins credentials configured in Jenkins itself and a custom maven settings file, so the workaround we found was to configure the additional repos (ksqldb staging repo and beta maven repo, needed for dependencies) in the cc-docker-ksql pom itself, and use the default Jenkins credentials here in the Jenkinsfile.